### DISCUSSION TITLE
Using Simulation Data for Analysis

### DISCUSSION THREAD

**Discussion by arashalizadeh** [snl-dakota/dakota#22] [Q&A] [answered]
Dear Dakota community

Is there a way to use simulation data and then run Dakota to do SA instead of running the solver to generate data?
In some cases simulations are complex or often run through 3rd party software. But the tabular "design of experiment" data is available as a csv or text file format. (input/output data)
If this is possible, could you please point me to the documentation that explains how to build a Dakota input file for this case?

Thanks
Arash

**Comment by jadamstephens** [comment, marked as answer]
Dakota cannot perform sensitivity analysis directly on samples it did not itself generate. So, you can't use Excel, Python, R, etc to generate a sample design, run the samples, and provide the data to Dakota for analysis.

However, our users quite often use Dakota in what we call "offline" mode. This involves tricking Dakota to generate and dump out parameter sets, which you then go off and run at your leisure. After the simulations are run, you create results files for Dakota to consume and run Dakota again to read them in. Call the first run of Dakota the sample-generation run, and the second run the results-comsumption run.

This is not a way that Dakota was really designed to be used, so there are a few tricks/tips/gotchas:

* You must set a `seed` for the sampling method and it must be the same in both the sample-generation run and the results-consuming run.
* By default, if Dakota sees a results file at the beginning of an evaluation, it will delete it. So, you need to add the `allow_existing_results` keyword to the `interface` section of your input file.
* You will need to write a dummy driver for Dakota to run. During the sample-generation run, the dummy driver should write dummy results back to Dakota. This is necessary because if Dakota can't find a valid results file, it'll abort. During the results-consumption run, the driver doesn't need to do anything. You've already generated and emplaced the real results files for Dakota to read.

In addition to the "offline" workflow, Dakota could be configured to import the data and use it to build a surrogate, which it then uses to perform sensitivty analysis. We have [an example](https://github.com/snl-dakota/dakota-examples/tree/master/official/surrogates/imported) that demonstrates that approach.
